src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -114,14 +115,26 @@ class AppTasksResource @Inject() (
       val deploymentF = taskKiller.killAndScale(pathId, findToKill, force)
       deploymentResult(result(deploymentF))
     } else {
-      reqToResponse(taskKiller.kill(pathId, findToKill, wipe)) {
-        tasks => ok(jsonObjString("tasks" -> tasks))
+      val response: Future[Response] = async {
+        val instances = await(taskKiller.kill(pathId, findToKill, wipe))
+        val healthStatuses = await(healthCheckManager.statuses(pathId))
+        val enrichedTasks: Seq[EnrichedTask] = instances.map { instance =>
+          val killedTask = instance.appTask // Should this rather be instance.taskMap[taskId]?
```

Comment:
It seems correct to me

Reply:
"Fun with TaskIds". Question: if the user requests a resident task with simplified id `app.task.uuid.6` to be killed (i.e. incarnation no 6), but the instance is at incarnation 7 (simplified id `app.task.uuid.7`), will `app.task.uuid.7` be killed?

:trollface: 

(I think this is something we don't need to fix in this API version)